File: src/main/java/uk/edu/populationfitness/output/TuningReader.java
package uk.edu.populationfitness.output;

import com.opencsv.CSVReader;
import uk.edu.populationfitness.Tuning;
import uk.edu.populationfitness.models.genes.Function;

import java.io.FileReader;
import java.io.IOException;

public class TuningReader {
    public static void read(Tuning tuning, String file) throws IOException{
        CSVReader reader = new CSVReader(new FileReader(file));

        // Read header
        reader.readNext();
        read(reader, tuning);
    }

    private static void read(CSVReader reader, Tuning tuning) throws IOException {
        String[] row = reader.readNext();
        tuning.function = Function.valueOf(row[0]);
        tuning.historic_fit = Double.parseDouble(row[1]);
        tuning.disease_fit = Double.parseDouble(row[2]);
        tuning.modern_fit = Double.parseDouble(row[3]);
        tuning.modern_breeding = Double.parseDouble(row[4]);
        tuning.size_of_genes = Integer.parseInt(row[5]);
        tuning.number_of_genes = Integer.parseInt(row[6]);
        tuning.mutations_per_gene = Double.parseDouble(row[7]);
        tuning.series_runs = Integer.parseInt(row[8]);
        tuning.parallel_runs = Integer.parseInt(row[9]);
    }
}

File: Scala/PopulationFitness/src/main/scala/uk/edu/populationfitness/models/output/TuningReader.scala
package uk.edu.populationfitness.models.output

import java.io.FileReader

import com.opencsv.CSVReader
import uk.edu.populationfitness.models.genes.FitnessFunction
import uk.edu.populationfitness.models.tuning.Tuning

object TuningReader {
  def read(file: String): Tuning = {
    val reader = new CSVReader(new FileReader(file))
    reader.readNext // read the header
    read(reader)
  }

  private def read(reader: CSVReader): Tuning = {
    val tuning = new Tuning
    val row = reader.readNext
    tuning.function = FitnessFunction.withName(row(0))
    tuning.historic_fit = row(1).toDouble
    tuning.disease_fit = row(2).toDouble
    tuning.modern_fit = row(3).toDouble
    tuning.modern_breeding = row(4).toDouble
    tuning.size_of_genes = row(5).toInt
    tuning.number_of_genes = row(6).toInt
    tuning.mutations_per_gene = row(7).toDouble
    tuning.series_runs = row(8).toInt
    tuning.parallel_runs = row(9).toInt
    tuning
  }
}
